### DISCUSSION TITLE
Recovering initial conditions based on measurements

### DISCUSSION THREAD

**Discussion by LucasJSch** [lululxvi/deepxde#1092] [Q&A]
Hello Dr. Lu Lu.
First of all, thanks for all the incredible effort and work that you put into this framework. Hats off to you!

My question: 
I'm looking to recover/learn the initial distribution of a field at a time instance `t0`, based on measurements at a final time instance `tf`. That distribution usually is not constant, and can be quite arbitrary.
I was wondering: How can I define within the framework a generic enough field to be learnt? Is there an example to base my code upon?

Another approach:
Based on the paper [Physics-informed neural networks for inverse problems in nano-optics and metamaterials](https://arxiv.org/pdf/1912.01085.pdf), there's the following diagram:
![Screenshot from 2022-12-21 21-18-28](https://user-images.githubusercontent.com/37854249/208986268-d82bc63a-761c-4052-9fb4-8bfbacc17dab.png)

This is exactly what I'm trying to achieve. The initial conditions are closely related to the material's profile, so if there's another way to formulate this problem without initial conditions it could also work for me. If the problem formulated in this diagram is already implemented, a link to the implementation would be very helpful.

**Comment by lululxvi** [comment]
If you know the solution at `tf`, then you can let `t <- -t`, so `tf` becomes IC.
